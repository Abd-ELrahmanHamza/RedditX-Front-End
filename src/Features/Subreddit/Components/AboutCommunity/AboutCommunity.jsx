import { BsShield, BsThreeDots } from "react-icons/bs";
import { Link } from "react-router-dom";
import { useIsModerator } from "Features/Subreddit/Contexts/IsModeratorProvider";

import {
  AboutCommunityContainer,
  Dots,
  ModTools,
  Text,
} from "./AboutCommunity.styled";
/**
 * header of Community Card
 *
 * @returns {React.Component}
 */
const AboutCommunity = () => {
  // let isMod = true;
  const {isMod} = useIsModerator();

  return (
    <AboutCommunityContainer>
      <Text>
        <h2>About Community</h2>
      </Text>
      <ModTools>
        {isMod && (
          <Link className="mod-tools" to="#">
            <span>
              <BsShield />
            </span>
            Mod Tools
          </Link>
        )}
        <Dots>
          <span>
            <BsThreeDots />
          </span>
        </Dots>
      </ModTools>
    </AboutCommunityContainer>
  );
};

export default AboutCommunity;
