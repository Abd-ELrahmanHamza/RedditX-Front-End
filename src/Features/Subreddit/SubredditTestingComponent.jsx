import { ThemeProvider } from "styled-components";
import defaultTheme from "Theme/defaultTheme";
import lightTheme from "Theme/lightTheme";
import { BrowserRouter as Router } from "react-router-dom";
import { SubRedditProvider } from "Features/Subreddit/Contexts/SubRedditProvider";
import { SubRedditIDProvider } from "./Contexts/SubRedditIDProvider";
import { IsModeratorProvider } from "Features/Subreddit/Contexts/IsModeratorProvider";
import { IsBannedProvider } from "Features/Subreddit/Contexts/IsBannedProvider";
import { IsMutedProvider } from "Features/Subreddit/Contexts/IsMutedProvider";

const theme = { ...defaultTheme, ...lightTheme };

/**
 * Function that wraps Children props inside a Browser Router and a Theme Provider
 * @returns {React.Component}The component is being returned.
 */
const SubredditTestingComponent = ({ children }) => {
  return (
    <Router>
      <ThemeProvider theme={theme}>
        <SubRedditProvider>
          <SubRedditIDProvider>
            <IsModeratorProvider>
              <IsBannedProvider>
                <IsMutedProvider>{children}</IsMutedProvider>
              </IsBannedProvider>
            </IsModeratorProvider>
          </SubRedditIDProvider>
        </SubRedditProvider>
      </ThemeProvider>
    </Router>
  );
};

export default SubredditTestingComponent;
