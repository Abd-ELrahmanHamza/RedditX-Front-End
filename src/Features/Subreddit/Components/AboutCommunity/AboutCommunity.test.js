import { screen, render } from "@testing-library/react";
import AboutCommunity from "./AboutCommunity";
import SubredditTestingComponent from "Features/Subreddit/SubredditTestingComponent";

describe("Header of Community card", () => {
  it("About Community should appear", () => {
    render(
      <SubredditTestingComponent>
        <AboutCommunity />
      </SubredditTestingComponent>
    );
    expect(screen.getByText(/About Community/i)).toBeInTheDocument();
  });
});
