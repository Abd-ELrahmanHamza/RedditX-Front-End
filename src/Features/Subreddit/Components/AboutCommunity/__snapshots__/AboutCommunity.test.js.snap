// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header of Community card should render without crashing 1`] = `ShallowWrapper {}`;
